pkg/api/mlflow/service/artifact/storage/s3.go

```diff
@@ -137,7 +138,11 @@ func (s S3) Get(artifactURI, itemPath string) (io.ReadCloser, error) {
 
 	resp, err := s.client.GetObject(context.TODO(), input)
 	if err != nil {
-		return nil, eris.Wrap(err, "error fetching object from s3")
+		notFoundRegexp := regexp.MustCompile("StatusCode.+404")
+		if notFoundRegexp.MatchString(err.Error()) {
+			return nil, eris.Wrap(err, PathError)
+		}
+		return nil, eris.Wrap(err, "unable to get object")
```

Comment:
please handle errors correctly — have a look at the [AWS SDK v2 docs](https://aws.github.io/aws-sdk-go-v2/docs/handling-errors/#api-error-responses) and check for the `*types.NoSuchKey` error

Reply:
I literally tried this locally with minio before typing the comment above